client/homebrew/editor/editor.jsx

```diff
@@ -102,6 +104,46 @@ const Editor = createClass({
 		}, 1);
 	},
 
+	addEditorPageNumbers : function(){
+		if(!this.refs.codeEditor) return;
+		if(this.state.view === 'text')  {
+			const codeMirror = this.refs.codeEditor.codeMirror;
+
+			const pageCountWidgets = document.getElementsByClassName('editor-page-count');
+			for (let x=pageCountWidgets.length - 1;x>=0;x--) pageCountWidgets[x].remove();
+
+			let editorPageCount = 2; // start counting from page 2
+
+			_.forEach(this.props.brew.text.split('\n'), (line, lineNumber)=>{
+				if(this.props.renderer == 'legacy') {
+					if(line.includes('\\page')) {
+						const testElement = Object.assign(document.createElement('div'), {
+							className   : 'editor-page-count',
+							textContent : editorPageCount
+						});
+						codeMirror.addWidget({ line: lineNumber, ch: 0 }, testElement);
+						testElement.style.top = `${parseInt(testElement.style.top) - 12.5}px`;
+						testElement.style.left = 'unset';
+						editorPageCount = editorPageCount + 1;
+					}
+				}
+
+				if(this.props.renderer == 'V3') {
+					if(line.match(/^\\page$/)){
+						const testElement = Object.assign(document.createElement('div'), {
+							className   : 'editor-page-count',
+							textContent : editorPageCount
+						});
+						codeMirror.addWidget({ line: lineNumber, ch: 0 }, testElement);
+						testElement.style.top = `${parseInt(testElement.style.top) - 12.5}px`;
+						testElement.style.left = 'unset';
+						editorPageCount = editorPageCount + 1;
+					}
+				}
+			}, []);
```

Comment:
```suggestion
			});
```

Since we aren't using `reduce` anymore I don't think we need this empty array that we were collecting line numbers into.